packages/rum-core/src/performance-monitoring/transaction-service.js

```diff
@@ -101,7 +101,36 @@ class TransactionService {
 
   createOptions(options) {
     const config = this._config.config
-    let presetOptions = { transactionSampleRate: config.transactionSampleRate }
+    let presetOptions = {
+      transactionSampleRate: config.transactionSampleRate
+    }
+    if (typeof config.transactionContextCallback === 'function') {
+      const tCC = function () {
+        let tags = {}
+        try {
+          tags = config.transactionContextCallback()
+        } catch (err) {
+          this._logger.error(
+            'Failed to execute transaction context callback',
+            err
+          )
+        }
+        return tags
+      }
+      presetOptions.transactionContextCallback = tCC.bind(this)
```

Comment:
I think is a good idea to bond a context to the callback function but in this case I would prefer to protect the RUM agent from external code by binding to a `null` context. This way the user defined callback has no access possibility to manipulate the transaction service via the `this` keyword.
```suggestion
      presetOptions.transactionContextCallback = tCC.bind(null)
```

Reply:
Binding to a null context seems to protect access against using the logger. I get the following error from transaction-service.spec.js if I try this change:
TypeError: Cannot read properties of null (reading '_logger')
Are there other changes needed to still access the logger? Or do we want to leave the binding set to this?

Reply:
I could also try binding to a new object which just defines the logger if that will be acceptable: e.g.
```
presetOptions.transactionContextCallback = tCC.bind({
  _logger: this._logger
})
```

Reply:
sorry, I missed you're using `this` to access the logger. In order to be able to `.bind(null)` you can refer to the logger within de closure like is done with `config`. 

```js
const config = this._config.config
const logger = this._logger // get a reference to the logger

...
if (...) {
  const tCC = function () {
    try {
      ...
    } catch (err) {
      logger.error('...', err) // then you can use it without `this.`
    }
  }
}

```

Reply:
I think we actually do not need to use the this keyword at all within the wrapper function